DevDaveFrame/wavepool
# Bug in News Post teaser

### To Verify

1. Clone/pull, run `pipenv install`, `pipenv shell`,
2. Run `python3 manage.py makemigrations` and `python3 manage.py migrate`
3. Run `python3 manage.py runserver`
4. Navigate to http://127.0.0.1:8000/admin/
5. Sign into the Django CMS Admin
6. Navigate to the "NewPost" admin page under "News"
7. Select any news post and enter text in the Teaser field at the bottom of the form
8. Save the news post
9. Navigate to http://127.0.0.1:8000/
10. Find the stories you selected either on the front page or in the archive
11. See the text you saved on your news post and the default teasers on other newsposts

### Commentary

This gave me more trouble than I anticipated. I decided to implement another model field which gave editors the option to set a custom teaser. That still left me the task of displaying an html-safe teaser by default for all the existing existing and future records, but I couldn't set the default using the field because I can't reference the body of an object that hasn't been created yet, so my quick fix was to add a default_teaser property that worked the same as the previous teaser implementation and add some filters in the template to at least partially clean the teaser, such that it won't break the UI. However, it will show an html tag that gets rescued by the `truncatechars_html` filter. By then I was running out of time, so I decided to leave it there for now.
